src/components/LogViewer/LogViewerBase.tsx

```diff
@@ -0,0 +1,91 @@
+import { Button } from '@patternfly/react-core';
+import { OutlinedPlayCircleIcon } from '@patternfly/react-icons';
+import { LogViewer as LogViewerPF } from '@patternfly/react-log-viewer';
+import { css } from '@patternfly/react-styles';
+import { ReactNode, useEffect, useState } from 'react';
+
+import styles from './LogViewer.module.css';
+
+interface ILogViewerBaseProps {
+  logViewerRef: React.RefObject<any>;
+  data: string | string[];
+  isStatic: boolean;
+  isPaused: boolean;
+  setIsPaused: (value: boolean) => void;
+  isFollowing: boolean;
+  areLinesWrapped: boolean;
+  toolbar?: ReactNode;
+}
+
+interface IOnScrollProps {
+  scrollOffsetToBottom: number;
+  scrollUpdateWasRequested: boolean;
+}
+
+export const LogViewerBase = ({
```

Comment:
Maybe consider using `React.memo` for child component, but I will leave it to you to decide whether to use memo or not.

Reply:
I think it is a good idea, I encapsulated it in the memo